services/QuillLMS/app/models/subscription.rb

```diff
@@ -244,16 +228,27 @@ def detach_district_admins
     end
   end
 
+  def renew_via_stripe
+    return ErrorNotifier.report(RenewalNilStripeCustomer.new, subscription_id: id) unless purchaser&.stripe_customer?
```

Comment:
Since you already have a `rescue` / `report` in the method, you could consider using:

```ruby
raise RenewalNilStripeCustomer unless purchaser&.stripe_customer?
# other code
rescue Stripe::InvalidRequestError, RenewalNilStripeCustomer => e

Reply:
I like this suggestion.  I was assuming that the error handling would be handled higher up, but this keeps it contained within the method which seems safer.